tests/dist/transport/test_point_to_point.cpp

```diff
@@ -69,6 +69,60 @@ TEST_CASE_METHOD(DistTestsFixture,
     }
 }
 
+TEST_CASE_METHOD(DistTestsFixture,
+                 "Test many in-order point-to-point messages",
+                 "[ptp][transport]")
+{
+    std::set<std::string> actualAvailable = sch.getAvailableHosts();
+    std::set<std::string> expectedAvailable = { getMasterIP(), getWorkerIP() };
+    REQUIRE(actualAvailable == expectedAvailable);
+
+    int appId = 222;
+    int groupId = 333;
+
+    // Set up this host's resources
+    // Make sure some functions execute remotely, some locally
+    int nLocalSlots = 1;
+    int nFuncs = 2;
+
+    faabric::HostResources res;
+    res.set_slots(nLocalSlots);
+    sch.setThisHostResources(res);
+
+    // Set up batch request
+    std::shared_ptr<faabric::BatchExecuteRequest> req =
+      faabric::util::batchExecFactory("ptp", "many-msg", nFuncs);
+
+    // Prepare expected decision
+    faabric::util::SchedulingDecision expectedDecision(appId, groupId);
+    std::vector<std::string> expectedHosts = { getMasterIP(), getWorkerIP() };
+
+    // Set up individual messages
+    for (int i = 0; i < nFuncs; i++) {
+        faabric::Message& msg = req->mutable_messages()->at(i);
+
+        msg.set_appid(appId);
+        msg.set_appidx(i);
+        msg.set_groupid(groupId);
+        msg.set_groupidx(i);
+
+        // Add to expected decision
+        expectedDecision.addMessage(expectedHosts.at(i), req->messages().at(i));
+    }
```

Comment:
Most of this function looks very similar to the other test above, could you refactor to share the logic rather than duplicate?